frontend/Plugin.js

```diff
@@ -0,0 +1,147 @@
+    (`path/to/tracking-sdk.${stage}.min.js`);
+    /* eslint-enable */
+    /* eslint-enable eslint-comments/no-unlimited-disable */
+
+    sgAnalytics('setConfig', {
+      shopNumber,
+      pushToken,
+      sgUserId: userId,
+      channel: 'app',
+    });
+
+    this.registerEvents();
+  }
+
+  /**
+   * Registers all event handlers.
+   */
+  registerEvents() {
+    this.register.pageview((data, rawData) => {
+      let sdkData = {
+        url: data.page.merchantUrl,
+        type: rawData.page.name,
+      };
+
+      if (sdkData.type === 'productDetails' && rawData.product) {
+        sdkData = {
+          ...sdkData,
+          id: rawData.product.uid,
+          product: {
+            number: rawData.product.uid || rawData.product.productNumber,
+            name: rawData.product.name,
+            currency: rawData.product.amount.currency,
+            price: rawData.product.amount.displayPrice ||
+             rawData.product.amount.grosss ||
+             rawData.product.amount.net,
+            priceStriked: rawData.product.amount.displayPriceStriked ||
+              rawData.product.amount.gross ||
+              rawData.product.amount.net,
+            identifiers: {
+              ean: rawData.product.identifiers.ean,
+            },
+          },
+        };
+      }
+
+      sgAnalytics('track', 'pageViewed', sdkData);
+      console.error('pageview', data, rawData, sdkData);
+    });
+
+    this.register.addToCart((data, rawData) => {
+      console.log(rawData);
+      const sdkData = {
+        products: rawData.products.map(product => ({
+          number: product.uid || product.productNumber,
+          name: product.name,
+          currency: product.amount.currency,
+          price: product.amount.displayPrice ||
+           product.amount.grosss ||
+           product.amount.net,
+          priceStriked: product.amount.displayPriceStriked ||
+            product.amount.gross ||
+            product.amount.net,
+          identifiers: {
+            ean: product.identifiers.ean,
+          },
+          quantity: product.quantity,
+        })),
+      };
+
+      sgAnalytics('track', 'productAddedToCart', sdkData);
+      console.error('addToCart', data, rawData, sdkData);
+    });
+
+    this.register.purchase((data, rawData) => {
+      console.error(data, rawData);
+      const sdkData = {
+        orderId: rawData.order.number,
+        shipping: {
+          type: rawData.order.shipping.name,
+          price: rawData.order.shipping.amount.gross,
+        },
+        coupons: rawData.order.coupons.map(coupon => ({
+          code: coupon.code,
+          savings: coupon.amount.gross,
+          currency: coupon.amount.currency,
+        })),
+        products: rawData.order.products.map(product => ({
```

Comment:
this could be outsourced into a function and used in multiple places

Reply:
done